simple-game-server-go/internal/game/allocated.go

```diff
@@ -120,3 +126,38 @@ func (g *Game) handleClient(client *net.TCPConn) {
 		}
 	}
 }
+
+// readyForPlayers waits 20s then reports that the game is ready for players.
+//
+// This is to simulate a game server waiting for any initialization to complete
+// before reporting that it is ready to accept players.
+//
+// You can optionally configure the timeout by setting the "readyTimeout"
+// configuration variable in the build configuration to a duration string
+// recognised by `time.ParseDuration`, e.g. "30s".
+func (g *Game) readyForPlayers() {
+	g.logger.Info("ready for players")
+
+	timeout := defaultReadyTimeout
+
+	if g.Config().Extra["readyTimeout"] != "" {
+		t, err := time.ParseDuration(g.Config().Extra["readyTimeout"])
+		if err != nil {
+			g.logger.WithError(err).Error("parsing ready timeout")
+		} else {
+			timeout = t
+		}
+	}
+
+	select {
+	case <-time.After(timeout):
+		g.logger.Info("ready timeout elapsed, reporting ready")
+	case <-g.OnDeallocate():
```

Comment:
We can't consume `OnDeallocate` here, as this could prevent the [current consumption](https://github.com/Unity-Technologies/multiplay-examples/blob/00ed8490855409ecdaab11c3d72e5053116814da/simple-game-server-go/internal/game/game.go#L90C1-L90C1) from being triggered. We should close a channel in there, and wait for the close here.

Reply:
Added a new Game.alloc channel to handle this